### ISSUE TITLE
Missing modules after installation

### ISSUE THREAD

**Issue by bpavie** [facebookresearch/sam3#252]
After installing SAM3 following the [instruction](https://github.com/facebookresearch/sam3?tab=readme-ov-file#installation) and try to run one of [jupyter notebook,](https://github.com/facebookresearch/sam3/blob/main/examples/sam3_video_predictor_example.ipynb) a lot of modules are missing.

On windows, here is what I had to install :

```bash
pip install einops
pip install triton-windows
pip install decord
pip install pycocotools
```

Note that for linux you can use the `pip install triton` directly

Best,
Benjamin

**Comment by mattiagaggi** [comment]
the issue is about triton-windows being different from triton. Made a fix

**Comment by ClayShoaf** [comment]
Add `pip install psutil` to that list, for me.

**Comment by syafiqmuda** [comment]
i try this using different PC, `pip install -e .` I just copy the code in readme and create `main.py`:

`main.py`:
```
import torch
#################################### For Image ####################################

from PIL import Image
from sam3.sam3.model_builder import build_sam3_image_model       # Ignore this since I create file outside sam3 main file directory
from sam3.sam3.model.sam3_image_processor import Sam3Processor   # Ignore this since I create file outside sam3 main file directory
# Load the model
model = build_sam3_image_model()
processor = Sam3Processor(model)
# Load an image
image = Image.open("<YOUR_IMAGE_PATH.jpg>")
inference_state = processor.set_image(image)
# Prompt the model with text
output = processor.set_text_prompt(state=inference_state, prompt="<YOUR_TEXT_PROMPT>")

# Get the masks, bounding boxes, and scores
masks, boxes, scores = output["masks"], output["boxes"], output["scores"]
```

and then I just run python main.py. it works on certain pc. is it possible because of missing module after installation. btw some PC already have necessary pip install in my conda environment. but if it missing I just install it. still not works. do I missing something here?

**Comment by AK2001** [comment]
Windows user also here, I had to install the following packages before I was able to run the basic tutorials:

```
ftfy==6.1.1
iopath==0.1.10
numpy==1.26.4
portalocker==3.2.0
pywin32==311
timm==1.0.26
setuptools==81.0.0
einops==0.8.2
triton-windows==3.6.0.post26
decord==0.6.0
pycocotools==2.0.11
```

**Comment by ethan-yrz** [comment]
Validated on a Gentoo-based HPC environment with Python 3.12 and CUDA:

- Pin `setuptools<81` because SAM 3 still imports `pkg_resources`.
- Install the missing image-import dependencies `einops`, Linux `triton`, and `psutil`.
- For `pycocotools`, this worked reliably:

```bash
export TMPDIR=/path/to/project-local/build-tmp
mkdir -p "$TMPDIR"
python -m pip install "Cython>=3.1,<4" wheel
python -m pip install --no-build-isolation --no-cache-dir "pycocotools==2.0.9"
```

`pycocotools` 2.0.10+ source enables `Py_LIMITED_API=0x030C0000` on CPython 3.12 and declares NumPy >=2 as a build requirement. Building it without isolation against SAM 3's NumPy 1.26/`numpy<2` constraint failed in NumPy headers (`PyUnicodeObject`). The official manylinux wheel was also rejected as unsupported by this Gentoo Python. Version 2.0.9 does not enable that limited-API branch and built successfully with Cython already installed.

Afterward, `pip check` and imports of `triton`, `pycocotools.mask`, `sam3.model_builder.build_sam3_image_model`, and `Sam3Processor` all passed.
